File: CarServiceGame/CarServiceGame.Domain/Entities/Garage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CarServiceGame.Domain.Entities
{
    public class Garage
    {
        public Guid GarageId { get; set; }

        public decimal CashBalance { get; set; }
        public int GarageLevel { get; set; }
        public List<Worker> EmployeedWorkers { get; set; }
        public List<RepairProcess> RepairProcesses { get; set; }

        public Garage()
        {
            
        }

        public Garage(Guid id, decimal balance, IEnumerable<Worker> workers, IEnumerable<RepairProcess> repairs = null)
        {
            GarageId = id;
            CashBalance = balance;
            EmployeedWorkers = new List<Worker>(workers);
            GarageLevel = 4;
            if (repairs != null)
            {
                if(repairs.Count()>4)
                    throw new ArgumentOutOfRangeException("repairs", "too many repair orders");
                RepairProcesses = new List<RepairProcess>(repairs);
            }
            else
                RepairProcesses = new List<RepairProcess>();
        }

        public void HireWorker(Worker worker)
        {
            EmployeedWorkers.Add(worker);
        }

        public void FireWorker(Worker worker)
        {
            EmployeedWorkers.Remove(worker);
        }

        public void FinishOrder(Guid orderId)
        {
            RepairProcesses.RemoveAll(x => x.Order.RepairOrderId == orderId);
        }

        public void CancelOrder(Guid orderId)
        {
            RepairProcesses.RemoveAll(x => x.Order.RepairOrderId == orderId);
        }

        public void UpgradeGarage()
        {
            GarageLevel += 2;
        }

        public void SetCashBalance(decimal balance)
        {
            CashBalance = balance;
        }

        public void AddRepairProcess(RepairProcess rp)
        {
            if(RepairProcesses == null)
                RepairProcesses = new List<RepairProcess>();

            RepairProcesses.Add(rp);
        }
    }
}

File: CarServiceGameWeb/src/app/model/garage.ts
import { RepairProcess } from "./repair-process";
import { Worker } from "./worker";

export class Garage {
    private _garageId: string; // TODO: find better data type for guid
    private _cashBalance: number;
    private _garageLevel: number;
    private _employeedWorkers: Array<Worker>
    private _repairProcesses: Array<RepairProcess>


    constructor(garageId: string, cashBalance: number, garageLevel: number) {
        this._garageId = garageId;
        this._cashBalance = cashBalance;
        this._garageLevel = garageLevel;
        this._employeedWorkers = new Array();
        this._repairProcesses = new Array();
    }


    get GarageId(): string {
        return this._garageId;
    }

    set GarageId(id: string) {
        this._garageId = id;
    }

    get CashBalance(): number {
        return this._cashBalance;
    }

    set CashBalance(balance: number) {
        this._cashBalance = balance;
    }

    get GarageLevel(): number {
        return this._garageLevel;
    }

    set GarageLevel(level: number) {
        this._garageLevel = level;
    }

    get EmployeedWorkers(): Array<Worker> {
        return this._employeedWorkers;
    }

    set EmployeedWorkers(workers: Array<Worker>) {
        this._employeedWorkers = workers;
    }

    get RepairProcesses(): Array<RepairProcess> {
        return this._repairProcesses
    }

    set RepairProcesses(processes: Array<RepairProcess>) {
        this._repairProcesses = processes;
    }

    addWorker(worker: Worker) {
        this._employeedWorkers.splice(0, 0, worker);
    }

    removeWorker(worker: Worker) {
        this._employeedWorkers = this._employeedWorkers.filter(x => x.WorkerId != worker.WorkerId);
    }
}